synfig-core/src/modules/mod_ffmpeg/mptr_ffmpeg.cpp

```diff
@@ -85,87 +74,35 @@ ffmpeg_mptr::seek_to(const Time& time)
 {
 	//if(frame<cur_frame || !file)
 	//{
-		if(file)
-		{
-#if defined(WIN32_PIPE_TO_PROCESSES)
-			_pclose(file);
-#elif defined(UNIX_PIPE_TO_PROCESSES)
-			fclose(file);
-			int status;
-			waitpid(pid,&status,0);
-#endif
-		}
+		pipe = nullptr;
 		
 		// FIXME: 24 fps is hardcoded now, but in fact we have to get it from canvas
 		//float position = (frame+1)/24; // ffmpeg didn't work with 0 frame
 		//float position = 1000/24; // ffmpeg didn't work with 0 frame
 		const std::string position = time.get_string(Time::FORMAT_NORMAL);
 
-#if defined(WIN32_PIPE_TO_PROCESSES)
+		OS::RunArgs args;
```

Comment:
In case if replace it with `std::vector<std::string> args;`:
Before:
```
OS::RunArgs args;
args.push_pair("-ss", position);
args.push_pair_filename("-i", identifier.filename);
args.push_pair("-vframes", "1");
args.push("-an");
args.push_pair("-f", "image2pipe");
args.push_pair("-vcodec", "ppm");
args.push("-");
```

After (variant 1):
```
std::vector<std::string> args;
args.push_back({"-ss", position, "-i", identifier.filename.escape(), "-vframes", "1", "-an", "-f", "image2pipe", "-vcodec", "ppm", "-"});
```

After (variant 2):
```
std::vector<std::string> args;
args.push_back({"-ss", position});
args.push_back({"-i", identifier.filename.escape()});
args.push_back({"-vframes", "1"});
args.push_back("-an");
args.push_back({"-f", "image2pipe"});
args.push_back({"-vcodec", "ppm"});
args.push_back("-");
```

The main difference is that the contributor knows what is going on without checking the `OS::RunArgs` class.

Reply:
The problem is that Windows needs filename quotes if it has spaces (because it uses a single string for the whole command) but Unix pipes must not have quotes (because it uses command string + arg array):

https://github.com/synfig/synfig/blob/cc59c1247fcdaa1c293bc1fc704922bd9348f7a1/synfig-core/src/modules/mod_ffmpeg/trgt_ffmpeg.cpp#L249-L253

https://github.com/synfig/synfig/blob/cc59c1247fcdaa1c293bc1fc704922bd9348f7a1/synfig-core/src/modules/mod_ffmpeg/trgt_ffmpeg.cpp#L294-L298

https://github.com/synfig/synfig/blob/cc59c1247fcdaa1c293bc1fc704922bd9348f7a1/synfig-core/src/modules/mod_ffmpeg/trgt_ffmpeg.cpp#L300-L306

What about create a method `push(const synfig::filesystem::Path& filename)` instead of `push_filename()` then?

Reply:
Agree

Reply:
Maybe name it `push_back` (because it adds data to the end of the list)?

Reply:
OK
I'll fix it after the current code is working.

Reply:
Done